### PR TITLE
SIMSBIOHUB-353/325: Survey form updates

### PR DESCRIPTION
**Pull request by al-rosenthal**
## Links to Jira Tickets

[SIMSBIOHUB-325](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-325)
[SIMSBIOHUB-353](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-353)

## Description of Changes

- Site selection strategies is now required for surveys 
- Removed field method from surveys

## Testing Notes

- You should be able to create and edit new and existing surveys without issue

### CODE REVIEW

File: api/src/paths/project/{projectId}/survey/create.ts [hunk 1]
```diff
@@ -202,7 +202,8 @@ POST.apiDoc = {
                   type: 'string'
                 },
                 field_method_id: {
```
Comment (curtisupshall) [inline api/src/paths/project/{projectId}/survey/create.ts]: It's probably in our best interest to remove and field mthod code entirely, and not just allow it to be null. Thoughts?

Comment (JeremyQuartech) [reply api/src/paths/project/{projectId}/survey/create.ts]: Yeah I agree, I think it could cause some confusing leaving this in here.

**Review by JeremyQuartech — COMMENTED** [review]
It might make sense to update the db seed file, adding a value for site selection strategy, seeing as it should be required, and removing the value for field method id. A deprecation comment in the db column for field_method_id could be useful too.

File: api/src/paths/project/{projectId}/survey/create.ts (hunk 1 above)
Comment (al-rosenthal) [reply api/src/paths/project/{projectId}/survey/create.ts]: good call, removed field method from the open api specs

**Comment by al-rosenthal** [comment]
> It might make sense to update the db seed file, adding a value for site selection strategy, seeing as it should be required, and removing the value for field method id. A deprecation comment in the db column for field_method_id could be useful too.

Good idea. I've removed the field method from the seed file and added the site strategy to the survey seed process

**Review by JeremyQuartech — APPROVED** [review]
Tested locally, everything looks good to me 👍

File: api/src/paths/project/{projectId}/survey/create.ts (hunk 1 above)
Comment (curtisupshall) [reply api/src/paths/project/{projectId}/survey/create.ts]: > good call, removed field method from the open api specs

What I mean is, we should probably remove any and all mention of `field_method` from the codebase. Example:

image

Comment (al-rosenthal) [reply api/src/paths/project/{projectId}/survey/create.ts]: I have since removed field method from the project 😅

**Review by curtisupshall — APPROVED** [review]
Code looks good to me. Tested this locally and everything works as expected. 👍

**Pull request merged**